pkg/machine/wsl/stubber.go

```diff
@@ -211,7 +218,14 @@ func (w WSLStubber) PostStartNetworking(mc *vmconfigs.MachineConfig, noInfo bool
 func (w WSLStubber) StartVM(mc *vmconfigs.MachineConfig) (func() error, func() error, error) {
 	dist := env.WithToolPrefix(mc.Name)
 
-	err := wslInvoke(dist, "/root/bootstrap")
+	var err error
+	if mc.Capabilities.GetForwardSockets() {
+		err = wslInvoke(dist, "/root/bootstrap")
+	} else {
+		// By using sudo, the script is run on a different tty device than the user (e.g user pts/0, sudo pts/1)
+		// this tricks WSL to not shut down the VM even if the user is not logged in because there is still an interactive operation running.
```

Comment:
`configureSystem` has some code to enable "linger" for the VM, but it’s disabled when macadam is used. Enabling linger may help with some of what you describe here, I don’t know if you experimented with it?

Reply:
yes, i tried almost everything. This hack was the only one that actually worked for me